markdowns/19_Text_to_Image_search_pipeline_with_MultiModal_Retriever.md

```diff
@@ -0,0 +1,190 @@
+---
+layout: tutorial
+colab: https://colab.research.google.com/github/deepset-ai/haystack-tutorials/blob/main/tutorials/19_Text_to_Image_search_pipeline_with_MultiModal_Retriever.ipynb
+toc: True
+title: "Text-to-Image search pipeline with MultiModal Retriever"
+last_updated: 2022-11-03
+level: "intermediate"
+weight: 95
+description: Use a MultiModalRetriever to build a cross modal search pipeline.
+category: "QA"
+aliases: ['/tutorials/multimodal']
+---
+    
+
+**Level**: Intermediate
+
+**Time to complete**: 20 minutes
+
+**Prerequisites**: Prepare the Colab environment (see links below).
+
+**Nodes Used**: InMemoryDocumentStore, MultiModalRetriever
+
+**Goal**: After completing this tutorial, you will have learned about the MultiModalRetriever, and built a simple retrieval pipeline that searches for relevant images given a text query.
+
+To make this pipeline work, we first need to embed the images from the dataset using a transformer model. These image embeddings are representative of the respective image's content. These embeddings lie in a high-dimensional vector space. Then we embed the text query using the same transformer model (i.e. OpenAI CLIP). Because we use the same transformer model, these text embeddings also exist in the same space as the image embeddings. Finally, we perform a nearest neighbor search to retrieve relevant images for our text query.
+
+Let's go and build a Text-to-Image search pipeline using a small animal dataset!
+
+## Preparing the Colab Environment
+
+- [Enable GPU Runtime in GPU](https://docs.haystack.deepset.ai/v5.2-unstable/docs/enable-gpu-runtime-in-colab)
+- [Check if GPU is Enabled](https://docs.haystack.deepset.ai/v5.2-unstable/docs/check-if-gpu-is-enabled)
+- [Set logging level to INFO](https://docs.haystack.deepset.ai/v5.2-unstable/docs/set-the-logging-level)
+
+
+```bash
+%%bash
+
+pip install --upgrade pip
+pip install farm-haystack[colab]
+```
+
+# Initializing the DocumentStore
+
+A DocumentStore contains Documents, which in this case are references to the images that Haystack will compare with your query. Here we are using the InMemoryDocumentStore since it requires no external dependencies. To learn more about the DocumentStore and the different types of external databases that we support, see [DocumentStore](https://docs.haystack.deepset.ai/docs/document_store).
```

Comment:
How about simplifying it a bit (have a look at [Voice and Tone](https://www.notion.so/deepsetai/Voice-and-Tone-Style-Guide-47952298c3f443e6973ae285ac89fb0a), we aim to reduce anxiety and time to value):

A DocumentStore stores references to the images that Haystack will compare with your query. But before it can do that, you need to initialize it. In this tutorial, you'll use the InMemoryDocumentStore. 
If you want to learn more, see [DocumentStore](https://docs.haystack.deepset.ai/docs/document_store).

(I'd say it's enough information for this tutorial, at this point they don't need to know what else a document store can store but I'm open to discuss this if you disagree :))

Reply:
I absolutely agree :)